### PR TITLE
added port autoneg test support for testbeds with AOS fanouts

### PR DESCRIPTION
**Pull request by antonptashnik**
### Description of PR

Summary: added port autoneg test support for testbeds with AOS fanouts
Fixes # (issue)

Port autoneg tests execute fanout commands via a common class FanoutHost which in turn delegates commands to OS specific impl. For now AOS class does not have methods required by port autoneg test. The PR resolves the issue
### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911

#### What is the motivation for this PR?
Enable port autoneg tests for testbeds with AOS fanout
#### How did you do it?
Implemented methods tests require
#### How did you verify/test it?
py.test --inventory=../ansible/lab,../ansible/veos --testbed_file=../ansible/testbed.csv --module-path=../ansible/library -v -rA --topology=t1,any platform_tests/test_auto_negotiation.py

### CODE REVIEW

**Comment by rlhui** [comment]
@prgeor, would you be able to help review

**Comment by SavchukRomanLv** [comment]
@prgeor @wangxin could you please take a look?

File: tests/common/devices/aos.py [hunk 1]
```diff
@@ -97,3 +109,86 @@ def exec_template(self, ansible_root, ansible_playbook, inventory, **kwargs):
...
+            # but in AOS autoneg activation and speeds advertisement is done via a single CLI cmd
+            # so this branch left nop intentionally
+            return True
+        speed = mb_to_speed(speed)
+        out = self.aos_config(
+                lines=['speed {}'.format(speed)],
+                parents='interface %s' % interface_name)
+        return not self._has_cli_cmd_failed(out)
+
+def speed_to_mb(speed):
```
Comment (wangxin) [inline tests/common/devices/aos.py:182]: Is `speed_gb_to_mg` a better name?

File: tests/common/devices/aos.py [hunk 2]
```diff
@@ -97,3 +109,86 @@ def exec_template(self, ansible_root, ansible_playbook, inventory, **kwargs):
...
+    res = re.search(r'(\d+)(\w)', speed)
+    if not res:
+        return speed
+    speed = res.groups()[0]
+    return speed + '000'
+
+def mb_to_speed(val):
```
Comment (wangxin) [inline tests/common/devices/aos.py:189]: Is `speed_mb_to_gb` a better name?

**Comment by antonptashnik** [comment]
@ZhaohuiS @wangxin please review

**Review by wangxin — APPROVED** [review]

**Pull request merged**